floodviz/static/js/hydrograph.js

```diff
@@ -1,149 +1,183 @@
-document.addEventListener("DOMContentLoaded", function (event) {
+(function () {
 	"use strict";
-
-	var data_path = FV.hydrograph_data_path;
-
-	// Set the dimensions of the canvas / graph
-	var margin = {top: 30, right: 20, bottom: 30, left: 50};
-	var width = FV.chart_dimensions.width - margin.left - margin.right;
-	var height = FV.chart_dimensions.height - margin.top - margin.bottom;
-
-	// Set the ranges
-	var x = d3.scaleTime().range([0, width]);
-	var y = d3.scaleLog().range([height, 0]);
-
-	// Define the voronoi
-	var voronoi = d3.voronoi()
-		.x(function (d) {
-			return x(d.time_mili);
-		})
-		.y(function (d) {
-			return y(d.value);
-		})
-		.extent([[-margin.left, -margin.top], [width + margin.right, height + margin.bottom]])
-
-	// Define the line
-	var line = d3.line()
-		.x(function (d) {
-			return x(d.time_mili);
-		})
-		.y(function (d) {
-			return y(d.value);
-		});
-
-	// Get the data
-	d3.json(data_path, function (error, data) {
-
-		//save the original data
-		var original_data = data;
-
-		update(data);
-
-		/**
-		 * @param {Array} data -- an array containing the json data to be drawn
-		 *
-		 * Draws the svg, scales the range of the data, draws the line for each site, and creates the Voronoi
-		 * tesselation and mouseover interactions, all based on the data set as it was passed in. Called as needed
-		 * when data changes (as in removal of a line).
-		 *
-		 */
-		function update(data) {
-
-			// Adds the svg canvas
-			var svg = d3.select("#hydrograph")
-				.append("svg")
-				.attr("width", width + margin.left + margin.right)
-				.attr("height", height + margin.top + margin.bottom)
-				.append("g")
-				.attr("transform",
-					"translate(" + margin.left + "," + margin.top + ")");
-
-			var graph_data = data.map(function(d) {
-				return  { "date": d.date, "key": d.key, "name": d.name, "time": d.time, "time_mili": d.time_mili,
-					"timezone": d.timezone, "value": Number(d.value) } ;
+	/**
+		* @param {Javascript Object} options - holds options for the configuration of the hydrograph
+		*	All keys are not optional.
+		*  Keys include: 	
+		*   'height' v(int) - height of the graph 
```

Comment:
don't need to change now, put typically I use @prop {Number} height - height of the graph. Indented under @param {Object} options. I put a link in the slack channel to the jsdoc standards we try to use. You should also document what the function is returning.